proposals/spi-proposal.md

```diff
@@ -112,113 +115,119 @@ MSBFIRST = 1
 | 2       | 1                     | 0                  |
 | 3       | 1                     | 1                  |
 
-
 #### maxSpeed
 
-The maximum speed of communication with the SPI device. For a SPI device rated up to
-5 MHz, use 5000000.
+The maximum speed of communication with the SPI device. For a SPI device rated
+up to 5 MHz, use 5000000.
+
+*For platforms that use a clock divider instead of a speed, pass the clock
+divider value instead.* 
 
 #### wordSize
 
-The size of a `word` in bits. Typically 8-bits (default). 0 = DEFAULT = 8-bits, 1 = 1 bit,
-2 = 2 bits, etc (limit is TBD).
+The size of a `word` in bits. Typically 8-bits (default). 0 = DEFAULT = 8-bits,
+1 = 1 bit, 2 = 2 bits, etc (limit is TBD).
 
-#### csPin / csPinOptions
+#### csPinOptions / csPin
 
-Specify a CS (chip select) pin to automatically toggle the pin rather than having to
-send a separate `DIGITAL_MESSAGE` command with every transfer.
+To enable automatic control of the csPin, set bit 0 (`CS_PIN_CONTROL`) to 1
+(enable). To manually control the csPin state with `DIGITAL_MESSAGE` commands,
+set `CS_PIN_CONTROL` to 0 (disable).
 
-The default behavior is to pull the CS pin LOW at the beginning of a transfer and toggle
-it at the end of the transfer. Setting bit 0 (`CS_ACTIVE_STATE`) will pull the CS pin HIGH
-at the start of the transfer. Set bit 1 (`CS_TOGGLE`) to handle the rare use case of
-toggling the CS pin between every word sent rather than at the end of the transaction.
+The default behavior is to pull the CS pin LOW at the beginning of a transaction
+and toggle it at the end of the transaction. Setting bit 1 (`CS_ACTIVE_STATE`)
+will pull the CS pin HIGH at the start of the transaction.
 
-The CS pin behavior can be further controlled by using the `csPinControl` byte of the
-`SPI_TRANSFER`, `SPI_WRITE` or `SPI_READ` messages.
+If `CS_PIN_CONTROL` is set to 0 (disable), the csPin parameter is ignored. Set
+the value to zero. The user will then need to manually control the csPin state
+using `DIGITAL_MESSAGE` commands.
 
 
+### SPI_BEGIN_TRANSACTION
```

Comment:
In Arduino-compatible implementations the CS pin would be automatically handled in the begin/end transaction commands. However the user would still need to send the begin transaction message, then the transfer, read or write message(s) then the end transaction message. Some of this added complexity can be masked with writeRegister, readRegister helper functions on the client side but there are times when the user would need to understand how to properly use the transaction framing messages.